### ISSUE TITLE
Email loop detected

### ISSUE THREAD

**Issue by aalaily** [osTicket/osTicket#3966]
### Description
We have recently upgraded osTicket to version  1.10.1 (9ae093d), on Apache/2.4.7:MySQL 5.7.19:PHP 5.5.9-1ubuntu4.20 and since the upgrade, we keep seeing the following error message in the logs every 3 minutes:

```
Email loop detected
It appears as though is being used as a forwarded or fetched email account and is also being used as a user / system account. Please correct the loop or seek technical assistance.
```
The error message is verbatim - it doesn't display which email address it is referring to.

![image](https://user-images.githubusercontent.com/2475759/30539903-a362f104-9c42-11e7-9c8a-d64679e5bd3b.png)

All email accounts have been configured correctly (including all agent email accounts)... however, the error persists.  The system properly alerts various individuals on ticket creation (via email and the web form).

Any thoughts?

Thanks.

**Comment by greezybacon** [comment]
As an example, the message suggest you're using "requests@company.tld" as an inbound address (so users might send an email there to request support), and you're also using the address as an outbound address (so you might send an email to that address to notify agents of a new ticket). The message is triggered when an outbound messages is received in an inbound mailbox.

**Comment by ntozier** [comment]
osTicket tries to detect if sending an email would result in a email loop.  Such as if you are using a mailing list for something something and sending the email would result in it sending to itself.  Or say if sending an email to a collaborator would email itself [in case one of the collaborators would email the system], etc.  Are you doing anything like @greezybacon or I describe?

**Comment by aalaily** [comment]
@greezybacon from what I can tell from the message, the system is suggesting that I'm using "" as an email address; hence the "It appears as though is being used as..." message.

Having said that, I do use "email1@company.com" to receive requests (which creates tickets) and "email1@company.com" to send notifications.

We've been using OS Ticket for months in this setup without ever seeing this error.  It just started showing up, every 3 minutes, after the upgrade.

**Comment by kclubok** [comment]
When you use email1@company.com to send notifications, is that in the To: or the From: header?  It shouldn't cause a problem if it is the From address.  If it is in the To address, then you probably would have had problems with email loops since you created that setup.

**Comment by aalaily** [comment]
@kclubok email1@company.com is being used in the "from:" address and "to:" is filled with the client/agent address.  I've double checked and none of the agent email addresses have "email1@company.com" as their default.

**Comment by greezybacon** [comment]
@aalaily what version were you using previously?

**Comment by aalaily** [comment]
@greezybacon version 1.10.0

**Comment by aalaily** [comment]
@greezybacon @kclubok @ntozier 

I found what the issue was and I'm posting it below in case this happens to someone else.

The email configuration in my installation of OSTicket is fine and there are actually no bugs with 1.10.1.  Here's what the problem was: one of my clients submitted a ticket via an email that did not accept incoming mail (and that had an autoresponder set).  Therefore, my email1@company.com was sending a confirmation email to a no-reply@company2.com.... this back and forth went on for until I deleted the offending ticket.  In addition, since the client email was no-reply@company2.com, I added a filter to my installation so that any messages coming in from "no-reply@*" does not generate an automatic response.

Thanks again, your various questions led me down the right path.

The upgrade had nothing to do with it... it was just bad luck to have upgraded and then immediately receive a ticket from a no-reply email.
